yarn/src/main/scala/org/apache/spark/deploy/yarn/YarnAllocator.scala

```diff
@@ -169,6 +172,24 @@ private[yarn] class YarnAllocator(
   }
 
   /**
+   * Init `executorIdCounter`
+   */
+  def initExecutorIdCounter(): Unit = {
+    val port = sparkConf.getInt("spark.yarn.am.port", 0)
+    SparkHadoopUtil.get.runAsSparkUser { () =>
+      val init = RpcEnv.create(
+        "executorIdCounterInit",
+        Utils.localHostName,
+        port,
+        sparkConf,
+        new SecurityManager(sparkConf))
+      val driver = init.setupEndpointRefByURI(driverUrl)
```

Comment:
Hi @andrewor14 , `driverRef` doesn't work in this case. Because, for my understanding, `driverRef` which endpoint name called `YarnScheduler` send message to `YarnSchedulerEndpoint` (or get message from `YarnSchedulerEndpoint`), while we should get max executorId from `CoarseGrainedSchedulerBackend.DriverEndpoint` which endpoint name called `CoarseGrainedScheduler`.

So, I think we should need a method to initialize `executorIdCounter`. And as you said, we should add huge comment huge comment related to SPARK-12864 to explain why we need to do this at this method. What‘s your opinion ?